ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionsQuestionnaireMapper.java

```diff
@@ -0,0 +1,223 @@
+
+            requireSupport.getHearingLoop()
+                .map(YesNoOption::name)
+                .map(CCDYesNoOption::valueOf)
+                .ifPresent(builder::hearingLoop);
+
+            requireSupport.getDisabledAccess()
+                .map(YesNoOption::name)
+                .map(CCDYesNoOption::valueOf)
+                .ifPresent(builder::disabledAccess);
+        };
+    }
+
+    private Consumer<ExpertRequest> toExpertRequest(
+        CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder
+    ) {
+        return expertRequest -> {
+            builder.expertEvidenceToExamine(expertRequest.getExpertEvidenceToExamine());
+            builder.reasonForExpertAdvice(expertRequest.getReasonForExpertAdvice());
+        };
+    }
+
+    private Consumer<Witness> toWitness(CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder) {
+        return witness -> {
+            builder.selfWitness(yesNoMapper.to(witness.getSelfWitness()));
+            witness.getNoOfOtherWitness().ifPresent(builder::numberOfOtherWitnesses);
+        };
+    }
+
+    private void toHearingLocation(
+        HearingLocation hearingLocation,
+        CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder
+    ) {
+        builder.hearingLocation(hearingLocation.getCourtName());
+        builder.hearingLocationSlug(hearingLocation.getHearingLocationSlug());
+        hearingLocation.getCourtAddress()
+            .ifPresent(address -> builder.hearingCourtAddress(addressMapper.to(address)));
+        builder.hearingLocationOption(CCDCourtLocationOption.valueOf(hearingLocation.getLocationOption().name()));
+        hearingLocation.getExceptionalCircumstancesReason().ifPresent(builder::exceptionalCircumstancesReason);
+    }
+
+    @Override
+    public DirectionsQuestionnaire from(CCDDirectionsQuestionnaire ccdDirectionsQuestionnaire) {
+        if (ccdDirectionsQuestionnaire == null) {
+            return null;
+        }
+
+        DirectionsQuestionnaire.DirectionsQuestionnaireBuilder builder = DirectionsQuestionnaire.builder();
+
+        builder.hearingLocation(extractHearingLocation(ccdDirectionsQuestionnaire));
+        builder.witness(extractWitness(ccdDirectionsQuestionnaire));
+        builder.expertRequest(extractExpertRequest(ccdDirectionsQuestionnaire));
+        builder.requireSupport(extractRequireSupport(ccdDirectionsQuestionnaire));
+
+        List<ExpertReport> expertReports = asStream(ccdDirectionsQuestionnaire.getExpertReports())
+            .map(expertRowMapper::from)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+
+        builder.expertReports(expertReports);
+
+        List<UnavailableDate> unavailableDates = asStream(ccdDirectionsQuestionnaire.getUnavailableDates())
+            .map(unavailableDateMapper::from)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+
+        builder.unavailableDates(unavailableDates);
```

Comment:
this statement can be combine with previous mapping

Reply:
This can not be combined with previous mapping, as the variable `unavailableDates` does not exist there